tests/components/anthemav/test_init.py

```diff
@@ -63,3 +47,32 @@ async def test_config_entry_not_ready(hass: HomeAssistant) -> None:
         await hass.config_entries.async_setup(mock_config_entry.entry_id)
         await hass.async_block_till_done()
         assert mock_config_entry.state is config_entries.ConfigEntryState.SETUP_RETRY
+
+
+async def test_anthemav_dispatcher_signal(
+    hass: HomeAssistant,
+    mock_connection_create: AsyncMock,
+    mock_anthemav: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test send update signal to dispatcher."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    # get the callback function that trigger the signal
+    avr_update_callback = mock_connection_create.call_args[1]["update_callback"]
+
+    mock_signal_received = Mock()
+
+    async_dispatcher_connect(
```

Comment:
We shouldn't interact with integration details like this. Check that an the state of an entity is updated after the callback is called instead.

Reply:
I added this test because codecov raised that I needed to test the update callback function. I wasn't sure how to test it. I'll try to simulate an entity change and assert for this. Thanks